source/core/index.ts

```diff
@@ -235,6 +235,10 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 			return;
 		}
 
+		this.options.signal?.addEventListener('abort', () => {
+			this.destroy(new Error('This operation was aborted.'));
```

Comment:
This needs the same as what I commented on the other AbortController PRs you did. 

https://github.com/sindresorhus/p-map/blob/a5faf425ad3f871d10b3a18e30ae10d1edc78311/index.js#L3-L31

And the error needs to be documented in the docs with the other errors.